lib/maths/CPeriodicityHypothesisTests.cc

```diff
@@ -1839,10 +1838,12 @@ bool CPeriodicityHypothesisTests::testPeriodWithScaling(const TTimeTimePr2Vec& w
     }
 
     // Compute the degrees of freedom given the alternative hypothesis.
-    TSizeVec repeats(calculateRepeats(windows, period_, m_BucketLength, values));
-    double b{static_cast<double>(
-        std::count_if(repeats.begin(), repeats.end(),
-                      [](std::size_t repeat) { return repeat > 0; }))};
+    double b{[&] {
```

Comment:
Can we avoid using the default capture mode here?

Reply:
Done in [5d3c3a3](https://github.com/elastic/ml-cpp/pull/721/commits/5d3c3a3b79319afb4949e5ad10666d9adf1036d3).